MatomoTracker/MatomoTracker.swift

```diff
@@ -399,6 +399,25 @@ extension MatomoTracker {
             dimension.index != index
         })
     }
+    
+    /// remove all without ignore
+    @objc public func removeAll(ignoreIndexs indexs: [NSNumber]? = nil) {
+        if let array = indexs, array.count > 0 {
+            
+            let tempArray = dimensions
+            dimensions.removeAll()
```

Comment:
Updating the instance variable `dimensions` is not thread safe.
But that is more a general problem and not unique for this PR.